sql/compression_defaults.sql

```diff
@@ -0,0 +1,291 @@
+    _schema_name NAME;
+    _hypertable_row _timescaledb_catalog.hypertable;
+    _orderby_names NAME[];
+    _dimension_names NAME[];
+    _first_index_attrs NAME[];
+    _orderby_clauses text[];
+    _confidence int;
+BEGIN
+    SELECT n.nspname, c.relname INTO STRICT _schema_name, _table_name
+    FROM pg_class c
+    INNER JOIN pg_namespace n ON (n.oid = c.relnamespace)
+    WHERE c.oid = relation;
+
+    SELECT * INTO STRICT _hypertable_row FROM _timescaledb_catalog.hypertable h WHERE h.table_name = _table_name AND h.schema_name = schema_name;
+
+    --start with the unique index columns minus the segment by columns
+    with index_attr as (
+        SELECT
+        a.attnum, min(a.pos) as pos
+        FROM
+             --is there a better way to pick the right unique index if there are multiple?
+            (select indkey, indnkeyatts from pg_catalog.pg_index where indisunique and indrelid = relation limit 1) i
+        INNER JOIN LATERAL
+            (select * from unnest(i.indkey) with ordinality) a(attnum, pos) ON (TRUE)
+        WHERE a.pos <= i.indnkeyatts
+        GROUP BY 1
+    )
+    SELECT
+      array_agg(a.attname ORDER BY i.pos) INTO _orderby_names
+    FROM
+      index_attr i
+    INNER JOIN
+      pg_attribute a on (a.attnum = i.attnum AND a.attrelid = relation)
+    WHERE
+      NOT(a.attname::text = ANY (segment_by_cols));
+
+    if _orderby_names is null then
+        _orderby_names := array[]::name[];
+        _confidence := 5;
+    else
+        _confidence := 8;
+    end if;
+
+    --add dimensions
+    SELECT
+      array_agg(d.column_name) INTO _dimension_names
+    FROM _timescaledb_catalog.dimension d
+    WHERE
+      d.hypertable_id = _hypertable_row.id
+      AND NOT(d.column_name::text = ANY (_orderby_names))
+      AND NOT(d.column_name::text = ANY (segment_by_cols));
+    _orderby_names := _orderby_names || _dimension_names;
+
+    RAISE NOTICE 'orderby names: % %', _orderby_names, _dimension_names;
+
+    --add the first attribute of any index
+    with index_attr as (
+        SELECT
+        a.attnum, min(a.pos) as pos
+        FROM
+            (select indkey, indnkeyatts from pg_catalog.pg_index where indrelid = relation) i
+        INNER JOIN LATERAL
+            (select * from unnest(i.indkey) with ordinality) a(attnum, pos) ON (TRUE)
+        WHERE a.pos = 1
+        GROUP BY 1
+    )
+    SELECT
+      array_agg(a.attname ORDER BY i.pos) INTO _first_index_attrs
+    FROM
+      index_attr i
+    INNER JOIN
+      pg_attribute a on (a.attnum = i.attnum AND a.attrelid = relation)
+    WHERE
+          NOT(a.attname::text = ANY (_orderby_names))
+      AND NOT(a.attname::text = ANY (segment_by_cols));
+
+    _orderby_names := _orderby_names || _first_index_attrs;
+
+    --add DESC to any dimensions
+    SELECT
+      array_agg(
+      CASE WHEN d.column_name IS NULL THEN
+        a.colname
+      ELSE
+        a.colname || ' DESC'
+      END ORDER BY pos) INTO STRICT _orderby_clauses
```

Comment:
i think we want to move away from DESC ordering as it has proven to be problematic